lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2110,103 +2110,41 @@ constructTransaction ctx genChange (ApiT wid) body = do
-                let outs = addressAmountToTxOut <$> content
-                utx <- liftHandler
-                    $ W.selectAssets @_ @_ @s @k wrk pp W.SelectAssetsParams
-                        { outputs = F.toList outs
-                        , pendingTxs
-                        , txContext = txCtx
-                        , utxoAvailableForInputs =
-                            UTxOSelection.fromIndex utxoAvailable
-                        , utxoAvailableForCollateral =
-                            UTxOIndex.toUTxO utxoAvailable
-                        , wallet
-                        }
-                        (const Prelude.id)
-                (FeeEstimation estMin _) <- liftHandler
-                    $ W.estimateFee
-                    $ W.selectAssets @_ @_ @s @k wrk pp W.SelectAssetsParams
-                        { outputs = F.toList outs
-                        , pendingTxs
-                        , txContext = txCtx
-                        , utxoAvailableForInputs =
-                            UTxOSelection.fromIndex utxoAvailable
-                        , utxoAvailableForCollateral =
-                            UTxOIndex.toUTxO utxoAvailable
-                        , wallet
-                        }
-                        getFee
-                sel <- liftHandler $
-                    W.assignChangeAddressesWithoutDbUpdate wrk wid genChange utx
-                sel' <- liftHandler
-                    $ W.selectAssets @_ @_ @s @k wrk pp W.SelectAssetsParams
-                        { outputs = F.toList outs
-                        , pendingTxs
-                        , txContext = txCtx
-                        , utxoAvailableForInputs =
-                            UTxOSelection.fromIndex utxoAvailable
-                        , utxoAvailableForCollateral =
-                            UTxOIndex.toUTxO utxoAvailable
-                        , wallet
-                        }
-                        transform
-                pure (sel, sel', estMin)
-            Just (ApiPaymentAll _) -> do
-                liftHandler $ throwE $ ErrConstructTxNotImplemented "ADP-909"
+        let runSelection outs = W.selectAssets @_ @_ @s @k wrk pp W.SelectAssetsParams
+                { outputs = outs
+                , pendingTxs
+                , txContext = txCtx
+                , utxoAvailableForInputs =
+                        UTxOSelection.fromIndex utxoAvailable
+                , utxoAvailableForCollateral =
+                        UTxOIndex.toUTxO utxoAvailable
+                , wallet
+                } transform
+
+        (sel, sel', fee) <- do
+            outs <- case (body ^. #payments) of
+                Nothing -> pure []
+                Just (ApiPaymentAddresses content) ->
+                    pure $ F.toList (addressAmountToTxOut <$> content)
+                Just (ApiPaymentAll _) -> do
+                    liftHandler $ throwE $ ErrConstructTxNotImplemented "ADP-909"
```

Comment:
ADP-909 is already closed... is it a message that's potentially displayed to the user? If so it needs to be reworded :sweat_smile: